src/jqLite.js

```diff
@@ -671,7 +671,10 @@ forEach({
       }
       return element.value;
     }
-    element.value = value;
+    // Workaround for Firefox validation #12102.
+    if (element.value != value) {
```

Comment:
Indeed, since https://github.com/angular/angular.js/blob/master/src/ng/directive/input.js#L1128 takes care of the _empty_ values (null, undefined), it should be fine :)
So, using `!==` sounds like the simplest (aka best) way out...